Source/DafnyLanguageServer.Test/DafnyLanguageServerTestBase.cs

```diff
@@ -152,6 +152,9 @@ ensures true
 #pragma warning disable VSTHRD110 // Observe result of async calls
       Server.Initialize(CancellationToken);
 #pragma warning restore VSTHRD110 // Observe result of async calls
+
+      Disposable.Add(Server);
```

Comment:
Nice! I started down this path, but it looks like I didn't go far enough. I didn't know about `Server.Services`.

Reply:
I did some debugging on the library source to find this solution, starting from the knowledge that Boogie's ExecutionEngine has a CheckerPool that needs to be disposed, and then walking up the ownership chain to figure out which objects need to be disposed as well.